src/main/java/org/fenixedu/academic/domain/person/MaritalStatus.java

```diff
@@ -52,4 +54,8 @@ public String getPresentationName() {
     public String getLocalizedName() {
         return BundleUtil.getString(Bundle.ENUMERATION, this.getClass().getName() + "." + name());
     }
+
+    public String toLocalizedString(Locale locale) {
```

Comment:
@joaofreitas-qubit here's a few changes:

1. Refactor the method to getLocalizedName(Locale locale)
2. We should always use the enum's FQDN so same values for different enums do not clash, so use `this.getClass().getName() + "." + name()` as key.
3. Finally you can rewrite the `getLocalizedName()` delegating to `getLocalizatedName(Locale locale)` the default locale to use can be retrieved in `I18N.getLocale()`